### PR TITLE
import of admin_centre for admin with get_objets_and_deps

### PR DESCRIPTION
**Pull request by kinnou02**
Admin without admin_centre are imported with coord at 0;0.

### CODE REVIEW

File: src/bin/osm2mimir.rs [hunk 1]
```diff
@@ -63,41 +62,43 @@ Options:
...
-            };
-            adm.coord.lat = node.lat;
-            adm.coord.lon = node.lon;
+
+    pub fn is_admin(&self, obj: &osmpbfreader::OsmObj) -> bool{
+        match *obj {
+            osmpbfreader::OsmObj::Relation(ref rel) => {
+                rel.tags.get("boundary").map_or(false, |v| {
+                    v == "administrative" && rel.tags.get("admin_level").map_or(false, |lvl| {
+                        self.admin_levels.contains(&lvl.parse::<u32>().unwrap())
```
Comment (antoine-de) [inline src/bin/osm2mimir.rs]: :imp:  unwrap

File: src/bin/osm2mimir.rs [hunk 2]
```diff
@@ -63,41 +62,43 @@ Options:
...
 ";
 
-fn update_coordinates(filename: &String, admins: &mut AdminsMap) {
-    if admins.is_empty() {
-        return;
+#[derive(Debug)]
+struct AdminMatcher {
+    admin_levels: HashSet<u32>,
+}
+impl AdminMatcher{
```
Comment (pbougue) [inline src/bin/osm2mimir.rs]: empty line is missing: I think you need a rust-fmt pass.

File: src/bin/osm2mimir.rs [hunk 3]
```diff
@@ -108,30 +109,32 @@ fn administartive_regions(filename: &String, levels: &HashSet<u32>) -> AdminsMap
                           relation.tags.get("admin_level"));
                     continue;
                 }
-                Some(ref l) if !levels.contains(&l) => continue,
+                //Some(ref l) if !levels.contains(&l) => continue,
```
Comment (antoine-de) [inline src/bin/osm2mimir.rs]: useless comment

File: src/bin/osm2mimir.rs [hunk 4]
```diff
@@ -63,41 +62,43 @@ Options:
...
-                None => continue,
-            };
-            adm.coord.lat = node.lat;
-            adm.coord.lon = node.lon;
+
+    pub fn is_admin(&self, obj: &osmpbfreader::OsmObj) -> bool{
+        match *obj {
+            osmpbfreader::OsmObj::Relation(ref rel) => {
+                rel.tags.get("boundary").map_or(false, |v| {
+                    v == "administrative" && rel.tags.get("admin_level").map_or(false, |lvl| {
```
Comment (pbougue) [inline src/bin/osm2mimir.rs]: I would avoid imbrication here if possible, as one is as important as the other

File: src/bin/osm2mimir.rs [hunk 5]
```diff
@@ -108,30 +109,32 @@ fn administartive_regions(filename: &String, levels: &HashSet<u32>) -> AdminsMap
...
-                    continue;
+            let coord_centre = relation.refs.iter().find(|rf| rf.role == "admin_centre").map_or(default_coord.clone(), |r| {
+                match objects.get(&r.member){
+                    Some(value) => {
+                        match value {
+                            &osmpbfreader::OsmObj::Node(ref node) => mimirsbrunn::Coord{lat: node.lat, lon: node.lon},
+                            _ => default_coord.clone(),
+                        }
+                    }
+                    None => default_coord.clone()
```
Comment (pbougue) [inline src/bin/osm2mimir.rs]: A method would be nice as we will probably compute centroid at some point

**Pull request merged**